kv/kvserver: TestStoreRangeSplitRaceUninitializedRHS failed [skipped]

[(kv/kvserver).TestStoreRangeSplitRaceUninitializedRHS failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2051052&tab=buildLog) on [master@0b65365fce6a9bdde6c611c92031877ade88a8b2](https://github.com/cockroachdb/cockroach/commits/0b65365fce6a9bdde6c611c92031877ade88a8b2):

```
Error types: (1) *contextutil.TimeoutError (2) context.deadlineExceededError
W200630 06:44:22.106836 368001 kv/kvserver/node_liveness.go:538  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 150ms
(1) operation "node liveness heartbeat" timed out after 150ms
Wraps: (2) context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) context.deadlineExceededError
W200630 06:44:22.123928 368205 kv/kvserver/node_liveness.go:538  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 150ms
(1) operation "node liveness heartbeat" timed out after 150ms
Wraps: (2) aborted during DistSender.Send: context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) *roachpb.internalError
E200630 06:44:22.161498 387578 kv/kvserver/replica_range_lease.go:339  [s1,r5/1:{/Systemtse-a��}] heartbeat failed on epoch increment
W200630 06:44:22.309647 368001 kv/kvserver/node_liveness.go:538  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 150ms
(1) operation "node liveness heartbeat" timed out after 150ms
Wraps: (2) aborted during DistSender.Send: context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) *roachpb.internalError
W200630 06:44:22.321225 368205 kv/kvserver/node_liveness.go:538  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 150ms
(1) operation "node liveness heartbeat" timed out after 150ms
Wraps: (2) aborted during DistSender.Send: context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) *roachpb.internalError
I200630 06:44:22.560846 387980 kv/kvserver/replica_command.go:413  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf7" [r43] (manual)
I200630 06:44:22.666342 387980 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2209eaa7] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.561514588,0 encountered previous write with future timestamp 1593499462.561514588,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:22.691763 387980 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2209eaa7] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.562763929,0 encountered previous write with future timestamp 1593499462.562763929,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:22.704542 387980 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2209eaa7] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.562763929,1 encountered previous write with future timestamp 1593499462.562763929,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:22.729960 387980 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2209eaa7] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.562763929,2 encountered previous write with future timestamp 1593499462.562763929,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:22.763078 387980 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2209eaa7] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.562763929,3 encountered previous write with future timestamp 1593499462.562763929,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:22.807223 387980 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2209eaa7] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.562763929,4 encountered previous write with future timestamp 1593499462.562763929,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
W200630 06:44:22.899552 367878 kv/kvserver/store_rebalancer.go:223  [s1,store-rebalancer] StorePool missing descriptor for local store
I200630 06:44:22.984716 388249 kv/kvserver/replica_command.go:413  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf6" [r44] (manual)
I200630 06:44:23.229653 388249 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=a516945f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.985206828,0 encountered previous write with future timestamp 1593499462.985206828,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:23.276346 388249 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=a516945f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.986198305,0 encountered previous write with future timestamp 1593499462.986198305,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:23.297203 388249 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=a516945f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.986198305,1 encountered previous write with future timestamp 1593499462.986198305,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:23.348812 388249 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=a516945f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.986198305,2 encountered previous write with future timestamp 1593499462.986198305,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
W200630 06:44:23.367335 368198 kv/kvserver/store_rebalancer.go:223  [s2,store-rebalancer] StorePool missing descriptor for local store
I200630 06:44:23.398408 388249 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=a516945f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.986198305,3 encountered previous write with future timestamp 1593499462.986198305,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:23.415034 388249 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=a516945f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593499462.986198305,4 encountered previous write with future timestamp 1593499462.986198305,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200630 06:44:23.723157 388400 util/stop/stopper.go:539  quiescing
I200630 06:44:23.728868 388399 util/stop/stopper.go:539  quiescing
I200630 06:44:23.729607 368107 kv/kvserver/queue.go:582  [s2,r5/2:{/Systemtse-a��}] rate limited in MaybeAdd (split): node unavailable; try another peer
W200630 06:44:23.729859 388515 kv/kvserver/intentresolver/intent_resolver.go:539  failed to cleanup transaction intents: failed to resolve intents: node unavailable; try another peer
I200630 06:44:23.730348 368107 kv/kvserver/queue.go:582  [s2,r5/2:{/Systemtse-a��}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200630 06:44:23.754751 368363 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200630 06:44:23.756233 368363 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200630 06:44:23.758384 368354 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200630 06:44:23.772860 388398 util/stop/stopper.go:539  quiescing
W200630 06:44:23.776950 368404 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200630 06:44:23.777818 368255 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200630 06:44:23.781477 368072 gossip/gossip.go:1518  [n2] no incoming or outgoing connections
W200630 06:44:23.801344 368319 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
I200630 06:44:23.806269 388398 util/stop/stopper.go:539  quiescing
I200630 06:44:23.808064 388398 util/stop/stopper.go:539  quiescing
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (58.04s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestStoreRangeSplitRaceUninitializedRHS PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStoreRangeSplitRaceUninitializedRHS.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestStoreRangeSplitRaceUninitializedRHS failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2061081&tab=buildLog) on [master@9e0a39fb605df4f03014be44fac297971a6b61f5](https://github.com/cockroachdb/cockroach/commits/9e0a39fb605df4f03014be44fac297971a6b61f5):

```
I200703 02:30:47.755190 403516 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=cc70b169] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743447.519256739,4 encountered previous write with future timestamp 1593743447.519256739,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.017197 403778 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xfa" [r40] (manual)
I200703 02:30:48.133082 403778 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=dfc561d2] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.031454235,0 encountered previous write with future timestamp 1593743448.031454235,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.146736 403778 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=dfc561d2] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.069571184,0 encountered previous write with future timestamp 1593743448.069571184,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.161285 403778 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=dfc561d2] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.069571184,1 encountered previous write with future timestamp 1593743448.069571184,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.177733 403778 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=dfc561d2] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.069571184,2 encountered previous write with future timestamp 1593743448.069571184,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.335868 403778 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=dfc561d2] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.069571184,3 encountered previous write with future timestamp 1593743448.069571184,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.374188 403778 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=dfc561d2] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.069571184,4 encountered previous write with future timestamp 1593743448.069571184,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.549566 403884 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf9" [r41] (manual)
I200703 02:30:48.631674 403884 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=d7189321] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.550203406,0 encountered previous write with future timestamp 1593743448.550203406,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.644989 403884 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=d7189321] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.551400425,0 encountered previous write with future timestamp 1593743448.551400425,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.656862 403884 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=d7189321] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.551400425,1 encountered previous write with future timestamp 1593743448.551400425,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.684693 403884 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=d7189321] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.551400425,2 encountered previous write with future timestamp 1593743448.551400425,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.708050 403884 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=d7189321] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.551400425,3 encountered previous write with future timestamp 1593743448.551400425,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.727165 403884 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=d7189321] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.551400425,4 encountered previous write with future timestamp 1593743448.551400425,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.869327 404141 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf8" [r42] (manual)
I200703 02:30:48.961618 404141 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=b6a32335] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.869887760,0 encountered previous write with future timestamp 1593743448.869887760,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:48.999645 404141 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=b6a32335] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.871195690,0 encountered previous write with future timestamp 1593743448.871195690,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.017198 404141 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=b6a32335] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.871195690,1 encountered previous write with future timestamp 1593743448.871195690,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.042252 404141 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=b6a32335] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.871195690,2 encountered previous write with future timestamp 1593743448.871195690,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.057743 404141 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=b6a32335] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.871195690,3 encountered previous write with future timestamp 1593743448.871195690,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.072632 404141 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=b6a32335] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743448.871195690,4 encountered previous write with future timestamp 1593743448.871195690,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.284564 404265 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf7" [r43] (manual)
I200703 02:30:49.371697 404265 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03a6ad5f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.285688066,0 encountered previous write with future timestamp 1593743449.285688066,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.385125 404265 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03a6ad5f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.286862850,0 encountered previous write with future timestamp 1593743449.286862850,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.397812 404265 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03a6ad5f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.286862850,1 encountered previous write with future timestamp 1593743449.286862850,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.418534 404265 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03a6ad5f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.286862850,2 encountered previous write with future timestamp 1593743449.286862850,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.444303 404265 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03a6ad5f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.286862850,3 encountered previous write with future timestamp 1593743449.286862850,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.469842 404265 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03a6ad5f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.286862850,4 encountered previous write with future timestamp 1593743449.286862850,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.638392 404586 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf6" [r44] (manual)
I200703 02:30:49.751753 404586 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0c03dafd] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.638966945,0 encountered previous write with future timestamp 1593743449.638966945,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.774930 404586 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0c03dafd] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.640225327,0 encountered previous write with future timestamp 1593743449.640225327,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.802251 404586 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0c03dafd] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.640225327,1 encountered previous write with future timestamp 1593743449.640225327,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.817955 404586 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0c03dafd] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.640225327,2 encountered previous write with future timestamp 1593743449.640225327,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.836337 404586 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0c03dafd] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.640225327,3 encountered previous write with future timestamp 1593743449.640225327,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:49.865081 404586 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0c03dafd] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593743449.640225327,4 encountered previous write with future timestamp 1593743449.640225327,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200703 02:30:50.037722 404624 util/stop/stopper.go:539  quiescing
I200703 02:30:50.039727 404623 util/stop/stopper.go:539  quiescing
W200703 02:30:50.041219 404730 kv/kvserver/intentresolver/intent_resolver.go:539  failed to cleanup transaction intents: failed to resolve intents: node unavailable; try another peer
I200703 02:30:50.061239 381393 kv/kvserver/queue.go:582  [s2,r5/2:{/Systemtse-a��}] rate limited in MaybeAdd (split): node unavailable; try another peer
I200703 02:30:50.061931 381393 kv/kvserver/queue.go:582  [s2,r5/2:{/Systemtse-a��}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200703 02:30:50.116252 404622 util/stop/stopper.go:539  quiescing
W200703 02:30:50.125666 381657 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200703 02:30:50.128175 381599 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200703 02:30:50.128522 381611 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200703 02:30:50.128417 381370 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200703 02:30:50.131567 381576 gossip/gossip.go:1504  [n2] no incoming or outgoing connections
I200703 02:30:50.132650 404622 util/stop/stopper.go:539  quiescing
I200703 02:30:50.134696 404622 util/stop/stopper.go:539  quiescing
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (46.47s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestStoreRangeSplitRaceUninitializedRHS PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStoreRangeSplitRaceUninitializedRHS.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestStoreRangeSplitRaceUninitializedRHS failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2063145&tab=buildLog) on [master@456a07cfc1e53b87abc7709052e54efb1450e758](https://github.com/cockroachdb/cockroach/commits/456a07cfc1e53b87abc7709052e54efb1450e758):

```
Wraps: (2) context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) context.deadlineExceededError
W200704 06:01:56.013964 360658 kv/kvserver/node_liveness.go:538  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 150ms
(1) operation "node liveness heartbeat" timed out after 150ms
Wraps: (2) context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) *roachpb.internalError
E200704 06:01:56.049599 375173 kv/kvserver/replica_range_lease.go:339  [s1,r5/1:{/Systemtse-a��}] heartbeat failed on epoch increment
I200704 06:01:56.058206 360440 kv/kvserver/node_liveness.go:528  [liveness-hb] heartbeat failed on epoch increment; retrying
I200704 06:01:56.170985 360440 kv/kvserver/node_liveness.go:897  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (context deadline exceeded)
W200704 06:01:56.171633 360440 kv/kvserver/node_liveness.go:538  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 150ms
(1) operation "node liveness heartbeat" timed out after 150ms
Wraps: (2) context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) context.deadlineExceededError
E200704 06:01:56.185359 375309 kv/kvserver/replica_range_lease.go:383  [s2,r39/2:a�{�-�}] mismatch incrementing epoch for liveness(nid:1 epo:1 exp:1593842515.993867656,0); actual is liveness(nid:1 epo:1 exp:1593842516.358652523,0)
I200704 06:01:56.473758 374842 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2347666b] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842514.710813107,1 encountered previous write with future timestamp 1593842514.710813107,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:56.521864 374842 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2347666b] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842514.710813107,2 encountered previous write with future timestamp 1593842514.710813107,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:56.566485 374842 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2347666b] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842514.710813107,3 encountered previous write with future timestamp 1593842514.710813107,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:56.619079 374842 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=2347666b] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842514.710813107,4 encountered previous write with future timestamp 1593842514.710813107,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:57.121862 375595 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf7" [r43] (manual)
I200704 06:01:57.310386 375595 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=326491e3] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842517.122572725,0 encountered previous write with future timestamp 1593842517.122572725,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:57.354297 360440 kv/kvserver/node_liveness.go:897  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (context deadline exceeded)
W200704 06:01:57.354958 360440 kv/kvserver/node_liveness.go:538  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 150ms
(1) operation "node liveness heartbeat" timed out after 150ms
Wraps: (2) context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) context.deadlineExceededError
I200704 06:01:57.486849 375595 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=326491e3] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842517.124141996,0 encountered previous write with future timestamp 1593842517.124141996,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:57.588192 375595 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=326491e3] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842517.124141996,1 encountered previous write with future timestamp 1593842517.124141996,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:57.633537 375595 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=326491e3] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842517.124141996,2 encountered previous write with future timestamp 1593842517.124141996,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:57.670833 375595 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=326491e3] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842517.124141996,3 encountered previous write with future timestamp 1593842517.124141996,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:57.705158 360439 gossip/gossip.go:1518  [n1] node has connected to cluster via gossip
I200704 06:01:57.708394 375595 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=326491e3] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842517.124141996,4 encountered previous write with future timestamp 1593842517.124141996,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:58.027514 375955 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf6" [r44] (manual)
I200704 06:01:58.156526 375955 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0ef38a65] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842518.035624219,1 encountered previous write with future timestamp 1593842518.035624219,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:58.223407 375955 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0ef38a65] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842518.036631225,0 encountered previous write with future timestamp 1593842518.036631225,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:58.239587 375955 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0ef38a65] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842518.036631225,1 encountered previous write with future timestamp 1593842518.036631225,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:58.393398 375955 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0ef38a65] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842518.036631225,2 encountered previous write with future timestamp 1593842518.036631225,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:58.424905 375955 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0ef38a65] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842518.036631225,3 encountered previous write with future timestamp 1593842518.036631225,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:58.476991 375955 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0ef38a65] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593842518.036631225,4 encountered previous write with future timestamp 1593842518.036631225,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200704 06:01:58.882569 376243 util/stop/stopper.go:539  quiescing
I200704 06:01:58.883265 376244 util/stop/stopper.go:539  quiescing
W200704 06:01:58.890204 376229 kv/kvserver/intentresolver/intent_resolver.go:539  failed to cleanup transaction intents: failed to resolve intents: node unavailable; try another peer
I200704 06:01:59.052722 376242 util/stop/stopper.go:539  quiescing
W200704 06:01:59.056287 360764 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200704 06:01:59.057607 360791 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200704 06:01:59.058695 360727 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200704 06:01:59.059252 360439 gossip/gossip.go:1504  [n1] no incoming or outgoing connections
W200704 06:01:59.059484 360720 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200704 06:01:59.071664 376242 util/stop/stopper.go:539  quiescing
I200704 06:01:59.073548 376242 util/stop/stopper.go:539  quiescing
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (41.77s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestStoreRangeSplitRaceUninitializedRHS PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStoreRangeSplitRaceUninitializedRHS.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestStoreRangeSplitRaceUninitializedRHS failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2064251&tab=buildLog) on [master@3e0de239121813ea4d47873388a2828a66d9edf7](https://github.com/cockroachdb/cockroach/commits/3e0de239121813ea4d47873388a2828a66d9edf7):

```
I200705 05:59:08.923371 392877 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xfa" [r40] (manual)
I200705 05:59:09.155224 392877 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0a5bb391] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928748.923797843,0 encountered previous write with future timestamp 1593928748.923797843,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.184645 392877 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0a5bb391] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928748.924681291,0 encountered previous write with future timestamp 1593928748.924681291,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.196765 392877 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0a5bb391] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928748.924681291,1 encountered previous write with future timestamp 1593928748.924681291,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.208592 392877 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0a5bb391] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928748.924681291,2 encountered previous write with future timestamp 1593928748.924681291,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.233625 392877 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0a5bb391] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928748.924681291,3 encountered previous write with future timestamp 1593928748.924681291,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.248044 392877 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=0a5bb391] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928748.924681291,4 encountered previous write with future timestamp 1593928748.924681291,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.543871 393173 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf9" [r41] (manual)
I200705 05:59:09.678967 393173 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=7c8c4b6e] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928749.544445887,0 encountered previous write with future timestamp 1593928749.544445887,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.734576 393173 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=7c8c4b6e] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928749.547262021,0 encountered previous write with future timestamp 1593928749.547262021,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.760805 393173 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=7c8c4b6e] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928749.547262021,1 encountered previous write with future timestamp 1593928749.547262021,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.772182 393173 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=7c8c4b6e] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928749.547262021,2 encountered previous write with future timestamp 1593928749.547262021,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.811072 393173 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=7c8c4b6e] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928749.547262021,3 encountered previous write with future timestamp 1593928749.547262021,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:09.832262 393173 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=7c8c4b6e] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928749.547262021,4 encountered previous write with future timestamp 1593928749.547262021,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.001637 393335 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf8" [r42] (manual)
I200705 05:59:10.077581 393335 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=41170154] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.002133110,0 encountered previous write with future timestamp 1593928750.002133110,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.099457 393335 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=41170154] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.003277716,0 encountered previous write with future timestamp 1593928750.003277716,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.115742 393335 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=41170154] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.003277716,1 encountered previous write with future timestamp 1593928750.003277716,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.129656 393335 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=41170154] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.003277716,2 encountered previous write with future timestamp 1593928750.003277716,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.157297 393335 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=41170154] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.003277716,3 encountered previous write with future timestamp 1593928750.003277716,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.169719 393335 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=41170154] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.003277716,4 encountered previous write with future timestamp 1593928750.003277716,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
W200705 05:59:10.338802 367141 kv/kvserver/store_rebalancer.go:223  [s1,store-rebalancer] StorePool missing descriptor for local store
I200705 05:59:10.407374 393524 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf7" [r43] (manual)
I200705 05:59:10.710223 393524 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ea4c0c7f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.407913521,0 encountered previous write with future timestamp 1593928750.407913521,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.726255 393524 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ea4c0c7f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.408983303,0 encountered previous write with future timestamp 1593928750.408983303,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.758643 393524 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ea4c0c7f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.408983303,1 encountered previous write with future timestamp 1593928750.408983303,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.823956 393524 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ea4c0c7f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.408983303,2 encountered previous write with future timestamp 1593928750.408983303,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.876334 393524 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ea4c0c7f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.408983303,3 encountered previous write with future timestamp 1593928750.408983303,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:10.898300 393524 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ea4c0c7f] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928750.408983303,4 encountered previous write with future timestamp 1593928750.408983303,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:11.086559 393737 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf6" [r44] (manual)
I200705 05:59:11.501395 393737 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03eb3bfc] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928751.086988548,0 encountered previous write with future timestamp 1593928751.086988548,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:11.584956 393737 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03eb3bfc] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928751.088142222,0 encountered previous write with future timestamp 1593928751.088142222,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:11.675869 393737 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03eb3bfc] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928751.088142222,1 encountered previous write with future timestamp 1593928751.088142222,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:11.686586 393737 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03eb3bfc] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928751.088142222,2 encountered previous write with future timestamp 1593928751.088142222,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:11.706883 393737 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03eb3bfc] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928751.088142222,3 encountered previous write with future timestamp 1593928751.088142222,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:11.726254 393737 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=03eb3bfc] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1593928751.088142222,4 encountered previous write with future timestamp 1593928751.088142222,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200705 05:59:11.918325 393932 util/stop/stopper.go:539  quiescing
I200705 05:59:11.918916 393931 util/stop/stopper.go:539  quiescing
W200705 05:59:11.920479 394021 kv/kvserver/intentresolver/intent_resolver.go:539  failed to cleanup transaction intents: failed to resolve intents: node unavailable; try another peer
I200705 05:59:11.940091 367323 kv/kvserver/queue.go:582  [s2,r5/2:{/Systemtse-a��}] rate limited in MaybeAdd (split): node unavailable; try another peer
I200705 05:59:11.940584 367323 kv/kvserver/queue.go:582  [s2,r5/2:{/Systemtse-a��}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200705 05:59:11.990662 393930 util/stop/stopper.go:539  quiescing
W200705 05:59:11.993529 367463 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200705 05:59:11.997551 367459 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200705 05:59:11.997916 367476 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200705 05:59:11.998136 367179 gossip/gossip.go:1504  [n2] no incoming or outgoing connections
W200705 05:59:11.998716 367401 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200705 05:59:12.000616 393930 util/stop/stopper.go:539  quiescing
I200705 05:59:12.002414 393930 util/stop/stopper.go:539  quiescing
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (69.75s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestStoreRangeSplitRaceUninitializedRHS PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStoreRangeSplitRaceUninitializedRHS.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestStoreRangeSplitRaceUninitializedRHS failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2065484&tab=buildLog) on [master@f61d73fe1e4773372f3800b715143305a3342ee7](https://github.com/cockroachdb/cockroach/commits/f61d73fe1e4773372f3800b715143305a3342ee7):

```
I200706 06:07:49.612534 395887 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=c35cbca4] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.444642434,0 encountered previous write with future timestamp 1594015669.444642434,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:49.639417 395887 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=c35cbca4] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.459182911,0 encountered previous write with future timestamp 1594015669.459182911,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:49.675478 395887 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=c35cbca4] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.459182911,1 encountered previous write with future timestamp 1594015669.459182911,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:49.721790 395887 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=c35cbca4] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.459182911,2 encountered previous write with future timestamp 1594015669.459182911,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:49.758826 395887 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=c35cbca4] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.459182911,3 encountered previous write with future timestamp 1594015669.459182911,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:49.772123 395887 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=c35cbca4] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.459182911,4 encountered previous write with future timestamp 1594015669.459182911,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:49.961213 396091 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf8" [r42] (manual)
I200706 06:07:50.374238 396091 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ca28b969] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.961800757,0 encountered previous write with future timestamp 1594015669.961800757,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:50.400957 396091 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ca28b969] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.964425580,0 encountered previous write with future timestamp 1594015669.964425580,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:50.420570 396091 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ca28b969] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.964425580,1 encountered previous write with future timestamp 1594015669.964425580,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:50.466434 396091 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ca28b969] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.964425580,2 encountered previous write with future timestamp 1594015669.964425580,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:50.488178 396091 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ca28b969] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.964425580,3 encountered previous write with future timestamp 1594015669.964425580,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:50.546079 396091 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=ca28b969] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015669.964425580,4 encountered previous write with future timestamp 1594015669.964425580,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:51.020224 396459 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf7" [r43] (manual)
I200706 06:07:51.182041 396459 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=aec1bb1a] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.025028700,0 encountered previous write with future timestamp 1594015671.025028700,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:51.233500 396459 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=aec1bb1a] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.026611395,0 encountered previous write with future timestamp 1594015671.026611395,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:51.286878 396459 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=aec1bb1a] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.026611395,1 encountered previous write with future timestamp 1594015671.026611395,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:51.323346 396459 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=aec1bb1a] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.026611395,2 encountered previous write with future timestamp 1594015671.026611395,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:51.344553 396459 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=aec1bb1a] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.026611395,3 encountered previous write with future timestamp 1594015671.026611395,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:51.365875 396459 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=aec1bb1a] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.026611395,4 encountered previous write with future timestamp 1594015671.026611395,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:51.838029 396732 kv/kvserver/replica_command.go:414  [s1,r5/1:{/Systemtse-a��}] initiating a split of this range at key "a\x87\xf6" [r44] (manual)
I200706 06:07:52.094651 396732 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=68df91b5] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.838634818,0 encountered previous write with future timestamp 1594015671.838634818,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:52.244299 396732 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=68df91b5] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.853865648,0 encountered previous write with future timestamp 1594015671.853865648,0 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:52.258721 396732 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=68df91b5] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.853865648,1 encountered previous write with future timestamp 1594015671.853865648,1 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:52.383369 396732 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=68df91b5] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.853865648,2 encountered previous write with future timestamp 1594015671.853865648,2 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:52.427216 396732 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=68df91b5] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.853865648,3 encountered previous write with future timestamp 1594015671.853865648,3 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:52.450607 396732 kv/kvserver/replica_evaluate.go:454  [s1,r5/1:{/Systemtse-a��},txn=68df91b5] test injecting error: ReadWithinUncertaintyIntervalError: read at time 1594015671.853865648,4 encountered previous write with future timestamp 1594015671.853865648,4 within uncertainty interval `t <= <nil>`; observed timestamps: []
I200706 06:07:52.791476 397047 util/stop/stopper.go:539  quiescing
I200706 06:07:52.792188 397048 util/stop/stopper.go:539  quiescing
W200706 06:07:52.793020 397044 kv/kvserver/intentresolver/intent_resolver.go:539  failed to cleanup transaction intents: failed to resolve intents: node unavailable; try another peer
I200706 06:07:52.813053 373679 kv/kvserver/queue.go:582  [s2,r40/2:a�{�-�}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200706 06:07:52.813260 373881 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200706 06:07:52.814692 373881 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200706 06:07:52.817690 373840 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
W200706 06:07:52.818916 373881 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
I200706 06:07:52.828953 397046 util/stop/stopper.go:539  quiescing
W200706 06:07:52.828840 373744 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n1,s1):1: no handler registered for (n2,s2):2
W200706 06:07:52.830440 373744 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n1,s1):1: no handler registered for (n2,s2):2
W200706 06:07:52.831423 373744 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n1,s1):1: no handler registered for (n2,s2):2
W200706 06:07:52.832176 373744 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n1,s1):1: no handler registered for (n2,s2):2
W200706 06:07:52.834758 373742 kv/kvserver/raft_transport.go:466  no handler found for store 1 in response range_id:43 from_replica:<node_id:2 store_id:2 replica_id:2 > to_replica:<node_id:1 store_id:1 replica_id:1 > union:<error:<message:"store 2 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:2 > > now:<> > > 
W200706 06:07:52.836050 373742 kv/kvserver/raft_transport.go:466  no handler found for store 1 in response range_id:43 from_replica:<node_id:2 store_id:2 replica_id:2 > to_replica:<node_id:1 store_id:1 replica_id:1 > union:<error:<message:"store 2 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:2 > > now:<> > > 
W200706 06:07:52.836798 373656 gossip/gossip.go:1504  [n2] no incoming or outgoing connections
W200706 06:07:52.836975 373742 kv/kvserver/raft_transport.go:466  no handler found for store 1 in response range_id:43 from_replica:<node_id:2 store_id:2 replica_id:2 > to_replica:<node_id:1 store_id:1 replica_id:1 > union:<error:<message:"store 2 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:2 > > now:<> > > 
W200706 06:07:52.838160 373634 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200706 06:07:52.840520 395098 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200706 06:07:52.841006 373851 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200706 06:07:52.843769 397046 util/stop/stopper.go:539  quiescing
I200706 06:07:52.846207 397046 util/stop/stopper.go:539  quiescing
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (88.52s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestStoreRangeSplitRaceUninitializedRHS PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStoreRangeSplitRaceUninitializedRHS.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/client_split_test.go
+++ b/pkg/kv/kvserver/client_split_test.go
@@ -45,7 +45,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/ts"
 	"github.com/cockroachdb/cockroach/pkg/ts/tspb"
@@ -81,13 +80,17 @@ func adminSplitArgs(splitKey roachpb.Key) *roachpb.AdminSplitRequest {
 func TestStoreRangeSplitAtIllegalKeys(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
 
-	cfg := kvserver.TestStoreConfig(nil)
-	cfg.TestingKnobs.DisableSplitQueue = true
-	cfg.TestingKnobs.DisableMergeQueue = true
-	store := createTestStoreWithConfig(t, stopper, cfg)
+	ctx := context.Background()
+	s, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	defer s.Stopper().Stop(ctx)
 
 	for _, key := range []roachpb.Key{
 		keys.Meta1Prefix,
@@ -98,7 +101,7 @@ func TestStoreRangeSplitAtIllegalKeys(t *testing.T) {
 		keys.SystemSQLCodec.TablePrefix(10 /* system descriptor ID */),
 	} {
 		args := adminSplitArgs(key)
-		_, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args)
+		_, pErr := kv.SendWrapped(context.Background(), s.DB().NonTransactionalSender(), args)
 		if !testutils.IsPError(pErr, "cannot split") {
 			t.Errorf("%q: unexpected split error %s", key, pErr)
 		}
@@ -111,16 +114,19 @@ func TestStoreSplitAbortSpan(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	manualClock := hlc.NewManualClock(2400 * time.Hour.Nanoseconds())
-	clock := hlc.NewClock(manualClock.UnixNano, time.Millisecond)
-	storeCfg := kvserver.TestStoreConfig(clock)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
 	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	left, middle, right := roachpb.Key("a"), roachpb.Key("b"), roachpb.Key("c")
 
@@ -166,10 +172,15 @@ func TestStoreSplitAbortSpan(t *testing.T) {
 	}
 
 	thresh := kvserverbase.TxnCleanupThreshold.Nanoseconds()
-	// Pick a non-gcable and gcable timestamp, respectively. Avoid the clock's
-	// exact timestamp because of unpredictable logical ticks.
-	tsFresh := hlc.Timestamp{WallTime: manualClock.UnixNano() - thresh + 1}
-	tsStale := hlc.Timestamp{WallTime: manualClock.UnixNano() - thresh - 1}
+	// Make sure this test doesn't run out of padding time if we significantly
+	// reduce TxnCleanupThreshold in the future for whatever reason.
+	require.Greater(t, thresh, int64(time.Minute))
+	// Pick a non-gcable and gcable timestamp, respectively.
+	// Use the current time for some non-expired abort span records.
+	// Note that the cleanup threshold is so large that while this test runs,
+	// these records won't expire.
+	tsFresh := hlc.Timestamp{WallTime: s.Clock().Now().WallTime}
+	tsStale := hlc.Timestamp{WallTime: s.Clock().Now().WallTime - thresh - 1}
 
 	args := []roachpb.Request{
 		populateAbortSpan(key(left, 1), tsFresh),
@@ -244,27 +255,35 @@ func TestStoreSplitAbortSpan(t *testing.T) {
 func TestStoreRangeSplitAtTablePrefix(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	key := keys.UserTableDataMin
 	args := adminSplitArgs(key)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), args); pErr != nil {
 		t.Fatalf("%q: split unexpected error: %s", key, pErr)
 	}
 
-	var desc descpb.TableDescriptor
+	var desc descpb.Descriptor
 	descBytes, err := protoutil.Marshal(&desc)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	// Update SystemConfig to trigger gossip.
-	if err := store.DB().Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+	if err := store.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		if err := txn.SetSystemConfigTrigger(true /* forSystemTenant */); err != nil {
 			return err
 		}
@@ -301,12 +320,20 @@ func TestStoreRangeSplitAtTablePrefix(t *testing.T) {
 func TestStoreRangeSplitInsideRow(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	// Manually create some the column keys corresponding to the table:
 	//
@@ -324,16 +351,16 @@ func TestStoreRangeSplitInsideRow(t *testing.T) {
 	}
 
 	// We don't care about the value, so just store any old thing.
-	if err := store.DB().Put(context.Background(), col1Key, "column 1"); err != nil {
+	if err := store.DB().Put(ctx, col1Key, "column 1"); err != nil {
 		t.Fatal(err)
 	}
-	if err := store.DB().Put(context.Background(), col2Key, "column 2"); err != nil {
+	if err := store.DB().Put(ctx, col2Key, "column 2"); err != nil {
 		t.Fatal(err)
 	}
 
 	// Split between col1Key and col2Key by splitting before col2Key.
 	args := adminSplitArgs(col2Key)
-	_, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args)
+	_, pErr := kv.SendWrapped(ctx, store.TestSender(), args)
 	if pErr != nil {
 		t.Fatalf("%s: split unexpected error: %s", col1Key, pErr)
 	}
@@ -362,27 +389,35 @@ func TestStoreRangeSplitInsideRow(t *testing.T) {
 func TestStoreRangeSplitIntents(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	// First, write some values left and right of the proposed split key.
 	pArgs := putArgs([]byte("c"), []byte("foo"))
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), pArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), pArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 	pArgs = putArgs([]byte("x"), []byte("bar"))
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), pArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), pArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
 	// Split the range.
 	splitKey := roachpb.Key("m")
 	args := adminSplitArgs(splitKey)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), args); pErr != nil {
 		t.Fatal(pErr)
 	}
 
@@ -393,7 +428,7 @@ func TestStoreRangeSplitIntents(t *testing.T) {
 	}
 	for _, key := range []roachpb.Key{keys.RangeDescriptorKey(roachpb.RKeyMin), keys.RangeDescriptorKey(splitKeyAddr)} {
 		if _, _, err := storage.MVCCGet(
-			context.Background(), store.Engine(), key, store.Clock().Now(), storage.MVCCGetOptions{},
+			ctx, store.Engine(), key, store.Clock().Now(), storage.MVCCGetOptions{},
 		); err != nil {
 			t.Errorf("failed to read consistent range descriptor for key %s: %+v", key, err)
 		}
@@ -435,26 +470,34 @@ func TestStoreRangeSplitIntents(t *testing.T) {
 func TestStoreRangeSplitAtRangeBounds(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	// Split range 1 at an arbitrary key.
 	key := roachpb.Key("a")
 	rngID := store.LookupReplica(roachpb.RKey(key)).RangeID
 	h := roachpb.Header{RangeID: rngID}
 	args := adminSplitArgs(key)
-	if _, pErr := kv.SendWrappedWith(context.Background(), store, h, args); pErr != nil {
+	if _, pErr := kv.SendWrappedWith(ctx, store, h, args); pErr != nil {
 		t.Fatal(pErr)
 	}
 	replCount := store.ReplicaCount()
 
 	// An AdminSplit request sent to the end of the old range
 	// should fail with a RangeKeyMismatchError.
-	_, pErr := kv.SendWrappedWith(context.Background(), store, h, args)
+	_, pErr := kv.SendWrappedWith(ctx, store, h, args)
 	if _, ok := pErr.GetDetail().(*roachpb.RangeKeyMismatchError); !ok {
 		t.Fatalf("expected RangeKeyMismatchError, found: %v", pErr)
 	}
@@ -463,7 +506,7 @@ func TestStoreRangeSplitAtRangeBounds(t *testing.T) {
 	// should succeed but no new ranges should be created.
 	newRng := store.LookupReplica(roachpb.RKey(key))
 	h.RangeID = newRng.RangeID
-	if _, pErr := kv.SendWrappedWith(context.Background(), store, h, args); pErr != nil {
+	if _, pErr := kv.SendWrappedWith(ctx, store, h, args); pErr != nil {
 		t.Fatal(pErr)
 	}
 
@@ -544,29 +587,31 @@ SELECT count(*), sum(value) FROM crdb_internal.node_metrics WHERE
 func TestStoreRangeSplitIdempotency(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithOpts(t,
-		testStoreOpts{
-			// This test was written before the test stores were able to start with
-			// more than one range and is not prepared to handle many ranges.
-			dontCreateSystemRanges: true,
-			cfg:                    &storeCfg},
-		stopper)
-	rangeID := roachpb.RangeID(1)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+
 	splitKey := roachpb.Key("m")
 	content := roachpb.Key("asdvb")
 
 	// First, write some values left and right of the proposed split key.
 	pArgs := putArgs([]byte("c"), content)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), pArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), pArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 	pArgs = putArgs([]byte("x"), content)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), pArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), pArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
@@ -578,7 +623,7 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 	lTxn := txn
 	lTxn.Sequence++
 	lIncArgs.Sequence = lTxn.Sequence
-	if _, pErr := kv.SendWrappedWith(context.Background(), store.TestSender(), roachpb.Header{
+	if _, pErr := kv.SendWrappedWith(ctx, store.TestSender(), roachpb.Header{
 		Txn: &lTxn,
 	}, lIncArgs); pErr != nil {
 		t.Fatal(pErr)
@@ -587,14 +632,16 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 	rTxn := txn
 	rTxn.Sequence++
 	rIncArgs.Sequence = rTxn.Sequence
-	if _, pErr := kv.SendWrappedWith(context.Background(), store.TestSender(), roachpb.Header{
+	if _, pErr := kv.SendWrappedWith(ctx, store.TestSender(), roachpb.Header{
 		Txn: &rTxn,
 	}, rIncArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
+	originalRepl := store.LookupReplica(roachpb.RKey(splitKey))
+	require.NotNil(t, originalRepl)
 	// Get the original stats for key and value bytes.
-	ms, err := stateloader.Make(rangeID).LoadMVCCStats(context.Background(), store.Engine())
+	ms, err := stateloader.Make(originalRepl.RangeID).LoadMVCCStats(ctx, store.Engine())
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -602,7 +649,7 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 
 	// Split the range.
 	args := adminSplitArgs(splitKey)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), args); pErr != nil {
 		t.Fatal(pErr)
 	}
 
@@ -619,20 +666,16 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 		}
 	}
 
-	repl := store.LookupReplica(roachpb.RKeyMin)
-	rngDesc := repl.Desc()
+	rngDesc := originalRepl.Desc()
 	newRng := store.LookupReplica([]byte("m"))
 	newRngDesc := newRng.Desc()
 	if !bytes.Equal(newRngDesc.StartKey, splitKey) || !bytes.Equal(splitKey, rngDesc.EndKey) {
 		t.Errorf("ranges mismatched, wanted %q=%q=%q", newRngDesc.StartKey, splitKey, rngDesc.EndKey)
 	}
-	if !bytes.Equal(newRngDesc.EndKey, roachpb.RKeyMax) || !bytes.Equal(rngDesc.StartKey, roachpb.RKeyMin) {
-		t.Errorf("new ranges do not cover KeyMin-KeyMax, but only %q-%q", rngDesc.StartKey, newRngDesc.EndKey)
-	}
 
 	// Try to get values from both left and right of where the split happened.
 	gArgs := getArgs([]byte("c"))
-	if reply, pErr := kv.SendWrapped(context.Background(), store.TestSender(), gArgs); pErr != nil {
+	if reply, pErr := kv.SendWrapped(ctx, store.TestSender(), gArgs); pErr != nil {
 		t.Fatal(pErr)
 	} else if replyBytes, pErr := reply.(*roachpb.GetResponse).Value.GetBytes(); pErr != nil {
 		t.Fatal(pErr)
@@ -640,7 +683,7 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 		t.Fatalf("actual value %q did not match expected value %q", replyBytes, content)
 	}
 	gArgs = getArgs([]byte("x"))
-	if reply, pErr := kv.SendWrappedWith(context.Background(), store.TestSender(), roachpb.Header{
+	if reply, pErr := kv.SendWrappedWith(ctx, store.TestSender(), roachpb.Header{
 		RangeID: newRng.RangeID,
 	}, gArgs); pErr != nil {
 		t.Fatal(pErr)
@@ -652,7 +695,7 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 
 	// Send out an increment request copied from above (same txn/sequence)
 	// which remains in the old range.
-	_, pErr := kv.SendWrappedWith(context.Background(), store.TestSender(), roachpb.Header{
+	_, pErr := kv.SendWrappedWith(ctx, store.TestSender(), roachpb.Header{
 		Txn: &lTxn,
 	}, lIncArgs)
 	if pErr != nil {
@@ -661,7 +704,7 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 
 	// Send out the same increment copied from above (same txn/sequence), but
 	// now to the newly created range (which should hold that key).
-	_, pErr = kv.SendWrappedWith(context.Background(), store.TestSender(), roachpb.Header{
+	_, pErr = kv.SendWrappedWith(ctx, store.TestSender(), roachpb.Header{
 		RangeID: newRng.RangeID,
 		Txn:     &rTxn,
 	}, rIncArgs)
@@ -671,12 +714,12 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 
 	// Compare stats of split ranges to ensure they are non zero and
 	// exceed the original range when summed.
-	left, err := stateloader.Make(rangeID).LoadMVCCStats(context.Background(), store.Engine())
+	left, err := stateloader.Make(originalRepl.RangeID).LoadMVCCStats(ctx, store.Engine())
 	if err != nil {
 		t.Fatal(err)
 	}
 	lKeyBytes, lValBytes := left.KeyBytes, left.ValBytes
-	right, err := stateloader.Make(newRng.RangeID).LoadMVCCStats(context.Background(), store.Engine())
+	right, err := stateloader.Make(newRng.RangeID).LoadMVCCStats(ctx, store.Engine())
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -688,10 +731,10 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 	if lValBytes == 0 || rValBytes == 0 {
 		t.Errorf("expected non-zero val bytes; got %d, %d", lValBytes, rValBytes)
 	}
-	if lKeyBytes+rKeyBytes <= keyBytes {
+	if lKeyBytes+rKeyBytes != keyBytes {
 		t.Errorf("left + right key bytes don't match; %d + %d <= %d", lKeyBytes, rKeyBytes, keyBytes)
 	}
-	if lValBytes+rValBytes <= valBytes {
+	if lValBytes+rValBytes != valBytes {
 		t.Errorf("left + right val bytes don't match; %d + %d <= %d", lValBytes, rValBytes, valBytes)
 	}
 }
@@ -704,14 +747,22 @@ func TestStoreRangeSplitIdempotency(t *testing.T) {
 func TestStoreRangeSplitStats(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	manual := hlc.NewManualClock(123)
-	storeCfg := kvserver.TestStoreConfig(hlc.NewClock(manual.UnixNano, time.Nanosecond))
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
 	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+
+	start := s.Clock().Now()
 
 	// Split the range after the last table data key.
 	keyPrefix := keys.SystemSQLCodec.TablePrefix(keys.MinUserDescID)
@@ -723,7 +774,7 @@ func TestStoreRangeSplitStats(t *testing.T) {
 	repl := store.LookupReplica(roachpb.RKey(keyPrefix))
 	// NOTE that this value is expected to change over time, depending on what
 	// we store in the sys-local keyspace. Update it accordingly for this test.
-	empty := enginepb.MVCCStats{LastUpdateNanos: manual.UnixNano()}
+	empty := enginepb.MVCCStats{LastUpdateNanos: start.WallTime}
 	if err := verifyRangeStats(store.Engine(), repl.RangeID, empty); err != nil {
 		t.Fatal(err)
 	}
@@ -738,14 +789,12 @@ func TestStoreRangeSplitStats(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := verifyRecomputedStats(snap, repl.Desc(), ms, manual.UnixNano()); err != nil {
+	if err := verifyRecomputedStats(snap, repl.Desc(), ms, start.WallTime); err != nil {
 		t.Fatalf("failed to verify range's stats before split: %+v", err)
 	}
 	if inMemMS := repl.GetMVCCStats(); inMemMS != ms {
 		t.Fatalf("in-memory and on-disk diverged:\n%+v\n!=\n%+v", inMemMS, ms)
 	}
-
-	manual.Increment(100)
 
 	// Split the range at approximate halfway point.
 	args = adminSplitArgs(midKey)
@@ -785,19 +834,18 @@ func TestStoreRangeSplitStats(t *testing.T) {
 	}
 
 	// Stats should both have the new timestamp.
-	now := manual.UnixNano()
-	if lTs := msLeft.LastUpdateNanos; lTs != now {
-		t.Errorf("expected left range stats to have new timestamp, want %d, got %d", now, lTs)
+	if lTs := msLeft.LastUpdateNanos; lTs < start.WallTime {
+		t.Errorf("expected left range stats to have new timestamp, want %d, got %d", start.WallTime, lTs)
 	}
-	if rTs := msRight.LastUpdateNanos; rTs != now {
-		t.Errorf("expected right range stats to have new timestamp, want %d, got %d", now, rTs)
+	if rTs := msRight.LastUpdateNanos; rTs < start.WallTime {
+		t.Errorf("expected right range stats to have new timestamp, want %d, got %d", start.WallTime, rTs)
 	}
 
 	// Stats should agree with recomputation.
-	if err := verifyRecomputedStats(snap, repl.Desc(), msLeft, now); err != nil {
+	if err := verifyRecomputedStats(snap, repl.Desc(), msLeft, s.Clock().PhysicalNow()); err != nil {
 		t.Fatalf("failed to verify left range's stats after split: %+v", err)
 	}
-	if err := verifyRecomputedStats(snap, replRight.Desc(), msRight, now); err != nil {
+	if err := verifyRecomputedStats(snap, replRight.Desc(), msRight, s.Clock().PhysicalNow()); err != nil {
 		t.Fatalf("failed to verify right range's stats after split: %+v", err)
 	}
 }
@@ -834,23 +882,17 @@ func TestStoreEmptyRangeSnapshotSize(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-
-	// Disable the replicate queue, the split queue, and the merge queue as we
-	// want to control both rebalancing, splits, and merges ourselves.
-	sc := kvserver.TestStoreConfig(nil)
-	sc.TestingKnobs.DisableReplicateQueue = true
-	sc.TestingKnobs.DisableSplitQueue = true
-	sc.TestingKnobs.DisableMergeQueue = true
-
-	mtc := &multiTestContext{storeConfig: &sc}
-	defer mtc.Stop()
-	mtc.Start(t, 2)
+	tc := testcluster.StartTestCluster(t, 2,
+		base.TestClusterArgs{
+			ReplicationMode: base.ReplicationManual,
+		})
+	defer tc.Stopper().Stop(ctx)
 
 	// Split the range after the last table data key to get a range that contains
 	// no user data.
 	splitKey := keys.SystemSQLCodec.TablePrefix(keys.MinUserDescID)
 	splitArgs := adminSplitArgs(splitKey)
-	if _, err := kv.SendWrapped(ctx, mtc.distSenders[0], splitArgs); err != nil {
+	if _, err := kv.SendWrapped(ctx, tc.Servers[0].DistSender(), splitArgs); err != nil {
 		t.Fatal(err)
 	}
 
@@ -861,7 +903,7 @@ func TestStoreEmptyRangeSnapshotSize(t *testing.T) {
 		headers []*kvserver.SnapshotRequest_Header
 	}{}
 	messageHandler := RaftMessageHandlerInterceptor{
-		RaftMessageHandler: mtc.stores[1],
+		RaftMessageHandler: tc.GetFirstStoreFromServer(t, 1),
 		handleSnapshotFilter: func(header *kvserver.SnapshotRequest_Header) {
 			// Each snapshot request is handled in a new goroutine, so we need
 			// synchronization.
@@ -870,12 +912,11 @@ func TestStoreEmptyRangeSnapshotSize(t *testing.T) {
 			messageRecorder.headers = append(messageRecorder.headers, header)
 		},
 	}
-	mtc.transport.Listen(mtc.stores[1].StoreID(), messageHandler)
+	tc.Servers[1].RaftTransport().Listen(tc.GetFirstStoreFromServer(t, 1).StoreID(), messageHandler)
 
 	// Replicate the newly-split range to trigger a snapshot request from store 0
 	// to store 1.
-	rangeID := mtc.stores[0].LookupReplica(roachpb.RKey(splitKey)).RangeID
-	mtc.replicateRange(rangeID, 1)
+	desc := tc.AddVotersOrFatal(t, splitKey, tc.Target(1))
 
 	// Verify that we saw at least one snapshot request,
 	messageRecorder.Lock()
@@ -884,7 +925,7 @@ func TestStoreEmptyRangeSnapshotSize(t *testing.T) {
 		t.Fatalf("expected at least one snapshot header, but got %d", a)
 	}
 	for i, header := range messageRecorder.headers {
-		if e, a := header.State.Desc.RangeID, rangeID; e != a {
+		if e, a := header.State.Desc.RangeID, desc.RangeID; e != a {
 			t.Errorf("%d: expected RangeID to be %d, but got %d", i, e, a)
 		}
 		if header.RangeSize != 0 {
@@ -907,13 +948,22 @@ func TestStoreEmptyRangeSnapshotSize(t *testing.T) {
 func TestStoreRangeSplitStatsWithMerges(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	manual := hlc.NewManualClock(123)
-	storeCfg := kvserver.TestStoreConfig(hlc.NewClock(manual.UnixNano, time.Nanosecond))
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
 	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableSplitQueue: true,
+				DisableMergeQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+
+	start := s.Clock().Now()
 
 	// Split the range after the last table data key.
 	keyPrefix := keys.SystemSQLCodec.TablePrefix(keys.MinUserDescID)
@@ -925,14 +975,13 @@ func TestStoreRangeSplitStatsWithMerges(t *testing.T) {
 	repl := store.LookupReplica(roachpb.RKey(keyPrefix))
 	// NOTE that this value is expected to change over time, depending on what
 	// we store in the sys-local keyspace. Update it accordingly for this test.
-	empty := enginepb.MVCCStats{LastUpdateNanos: manual.UnixNano()}
+	empty := enginepb.MVCCStats{LastUpdateNanos: start.WallTime}
 	if err := verifyRangeStats(store.Engine(), repl.RangeID, empty); err != nil {
 		t.Fatal(err)
 	}
 
 	// Write random TimeSeries data.
 	midKey := writeRandomTimeSeriesDataToRange(t, store, repl.RangeID, keyPrefix)
-	manual.Increment(100)
 
 	// Split the range at approximate halfway point.
 	args = adminSplitArgs(midKey)
@@ -955,19 +1004,18 @@ func TestStoreRangeSplitStatsWithMerges(t *testing.T) {
 	}
 
 	// Stats should both have the new timestamp.
-	now := manual.UnixNano()
-	if lTs := msLeft.LastUpdateNanos; lTs != now {
-		t.Errorf("expected left range stats to have new timestamp, want %d, got %d", now, lTs)
+	if lTs := msLeft.LastUpdateNanos; lTs < start.WallTime {
+		t.Errorf("expected left range stats to have new timestamp, want %d, got %d", start.WallTime, lTs)
 	}
-	if rTs := msRight.LastUpdateNanos; rTs != now {
-		t.Errorf("expected right range stats to have new timestamp, want %d, got %d", now, rTs)
+	if rTs := msRight.LastUpdateNanos; rTs < start.WallTime {
+		t.Errorf("expected right range stats to have new timestamp, want %d, got %d", start.WallTime, rTs)
 	}
 
 	// Stats should agree with recomputation.
-	if err := verifyRecomputedStats(snap, repl.Desc(), msLeft, now); err != nil {
+	if err := verifyRecomputedStats(snap, repl.Desc(), msLeft, s.Clock().PhysicalNow()); err != nil {
 		t.Fatalf("failed to verify left range's stats after split: %+v", err)
 	}
-	if err := verifyRecomputedStats(snap, replRight.Desc(), msRight, now); err != nil {
+	if err := verifyRecomputedStats(snap, replRight.Desc(), msRight, s.Clock().PhysicalNow()); err != nil {
 		t.Fatalf("failed to verify right range's stats after split: %+v", err)
 	}
 }
@@ -1020,12 +1068,20 @@ func fillRange(
 func TestStoreZoneUpdateAndRangeSplit(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	storeCfg := kvserver.TestStoreConfig(nil /* clock */)
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
-	config.TestingSetupZoneConfigHook(stopper)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+	config.TestingSetupZoneConfigHook(s.Stopper())
 
 	const maxBytes = 1 << 16
 	// Set max bytes.
@@ -1081,12 +1137,20 @@ func TestStoreZoneUpdateAndRangeSplit(t *testing.T) {
 func TestStoreRangeSplitWithMaxBytesUpdate(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	storeCfg := kvserver.TestStoreConfig(nil /* clock */)
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
-	config.TestingSetupZoneConfigHook(stopper)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+	config.TestingSetupZoneConfigHook(s.Stopper())
 
 	origRng := store.LookupReplica(roachpb.RKeyMin)
 
@@ -1153,15 +1217,14 @@ func TestStoreRangeSplitBackpressureWrites(t *testing.T) {
 			var activateSplitFilter int32
 			splitKey := roachpb.RKey(keys.UserTableDataMin)
 			splitPending, blockSplits := make(chan struct{}), make(chan struct{})
-			storeCfg := kvserver.TestStoreConfig(nil)
+
 			// Set maxBytes to something small so we can exceed the maximum split
 			// size without adding 2x64MB of data.
 			const maxBytes = 1 << 16
-			storeCfg.DefaultZoneConfig.RangeMaxBytes = proto.Int64(maxBytes)
-			storeCfg.TestingKnobs.DisableGCQueue = true
-			storeCfg.TestingKnobs.DisableMergeQueue = true
-			storeCfg.TestingKnobs.DisableSplitQueue = true
-			storeCfg.TestingKnobs.TestingRequestFilter =
+			zoneConfig := zonepb.DefaultZoneConfig()
+			zoneConfig.RangeMaxBytes = proto.Int64(maxBytes)
+
+			testingRequestFilter :=
 				func(_ context.Context, ba roachpb.BatchRequest) *roachpb.Error {
 					for _, req := range ba.Requests {
 						if cPut, ok := req.GetInner().(*roachpb.ConditionalPutRequest); ok {
@@ -1180,9 +1243,23 @@ func TestStoreRangeSplitBackpressureWrites(t *testing.T) {
 				}
 
 			ctx := context.Background()
-			stopper := stop.NewStopper()
-			defer stopper.Stop(ctx)
-			store := createTestStoreWithConfig(t, stopper, storeCfg)
+			serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+				Knobs: base.TestingKnobs{
+					Server: &server.TestingKnobs{
+						DefaultZoneConfigOverride: &zoneConfig,
+					},
+					Store: &kvserver.StoreTestingKnobs{
+						DisableGCQueue:       true,
+						DisableMergeQueue:    true,
+						DisableSplitQueue:    true,
+						TestingRequestFilter: testingRequestFilter,
+					},
+				},
+			})
+			s := serv.(*server.TestServer)
+			defer s.Stopper().Stop(ctx)
+			store, err := s.Stores().GetStore(s.GetFirstStoreID())
+			require.NoError(t, err)
 
 			// Split at the split key.
 			sArgs := adminSplitArgs(splitKey.AsRawKey())
@@ -1206,7 +1283,7 @@ func TestStoreRangeSplitBackpressureWrites(t *testing.T) {
 			// it gets blocked in the response filter.
 			if tc.splitOngoing {
 				atomic.StoreInt32(&activateSplitFilter, 1)
-				if err := stopper.RunAsyncTask(ctx, "force split", func(_ context.Context) {
+				if err := s.Stopper().RunAsyncTask(ctx, "force split", func(_ context.Context) {
 					store.SetSplitQueueActive(true)
 					if err := store.ForceSplitScanAndProcess(); err != nil {
 						log.Fatalf(ctx, "%v", err)
@@ -1288,11 +1365,11 @@ func TestStoreRangeSplitBackpressureWrites(t *testing.T) {
 func TestStoreRangeSystemSplits(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
 	// Intentionally leave the merge queue enabled. This indirectly tests that the
 	// merge queue respects these split points.
-	store, _ := createTestStore(t, stopper)
+	ctx := context.Background()
+	s, _, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer s.Stopper().Stop(ctx)
 
 	userTableMax := keys.MinUserDescID + 4
 	var exceptions map[int]struct{}
@@ -1304,17 +1381,17 @@ func TestStoreRangeSystemSplits(t *testing.T) {
 	//   - descriptor IDs are used to determine split keys
 	//   - the write triggers a SystemConfig update and gossip
 	// We should end up with splits at each user table prefix.
-	if err := store.DB().Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+	if err := s.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		if err := txn.SetSystemConfigTrigger(true /* forSystemTenant */); err != nil {
 			return err
 		}
 		descTablePrefix := keys.SystemSQLCodec.TablePrefix(keys.DescriptorTableID)
 		kvs, _ /* splits */ := schema.GetInitialValues()
-		for _, kv := range kvs {
-			if !bytes.HasPrefix(kv.Key, descTablePrefix) {
+		for _, akv := range kvs {
+			if !bytes.HasPrefix(akv.Key, descTablePrefix) {
 				continue
 			}
-			if err := txn.Put(ctx, kv.Key, &kv.Value); err != nil {
+			if err := txn.Put(ctx, akv.Key, &akv.Value); err != nil {
 				return err
 			}
 		}
@@ -1363,7 +1440,7 @@ func TestStoreRangeSystemSplits(t *testing.T) {
 		expKeys = append(expKeys, testutils.MakeKey(keys.Meta2Prefix, roachpb.RKeyMax))
 
 		testutils.SucceedsSoon(t, func() error {
-			rows, err := store.DB().Scan(context.Background(), keys.Meta2Prefix, keys.MetaMax, 0)
+			rows, err := s.DB().Scan(context.Background(), keys.Meta2Prefix, keys.MetaMax, 0)
 			if err != nil {
 				return err
 			}
@@ -1383,7 +1460,7 @@ func TestStoreRangeSystemSplits(t *testing.T) {
 	// Write another, disjoint (+3) descriptor for a user table.
 	userTableMax += 3
 	exceptions = map[int]struct{}{userTableMax - 1: {}, userTableMax - 2: {}}
-	if err := store.DB().Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+	if err := s.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		if err := txn.SetSystemConfigTrigger(true /* forSystemTenant */); err != nil {
 			return err
 		}
@@ -1903,12 +1980,20 @@ func TestStoreSplitOnRemovedReplica(t *testing.T) {
 func TestStoreSplitGCThreshold(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	leftKey := roachpb.Key("a")
 	splitKey := roachpb.Key("b")
@@ -1916,11 +2001,11 @@ func TestStoreSplitGCThreshold(t *testing.T) {
 	content := []byte("test")
 
 	pArgs := putArgs(leftKey, content)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), pArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), pArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 	pArgs = putArgs(rightKey, content)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), pArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), pArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
@@ -1934,12 +2019,12 @@ func TestStoreSplitGCThreshold(t *testing.T) {
 		},
 		Threshold: specifiedGCThreshold,
 	}
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), gcArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), gcArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
 	args := adminSplitArgs(splitKey)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), args); pErr != nil {
 		t.Fatal(pErr)
 	}
 
@@ -1950,7 +2035,7 @@ func TestStoreSplitGCThreshold(t *testing.T) {
 		t.Fatalf("expected RHS's GCThreshold is equal to %v, but got %v", specifiedGCThreshold, gcThreshold)
 	}
 
-	repl.AssertState(context.Background(), store.Engine())
+	repl.AssertState(ctx, store.Engine())
 }
 
 // TestStoreRangeSplitRaceUninitializedRHS reproduces #7600 (before it was
@@ -1961,17 +2046,6 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	// Skipping as part of test-infra-team flaky test cleanup.
-	skip.WithIssue(t, 50809)
-
-	mtc := &multiTestContext{}
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	// An aggressive tick interval lets groups communicate more and thus
-	// triggers test failures much more reliably. We can't go too aggressive
-	// or race tests never make any progress.
-	storeCfg.RaftTickInterval = 50 * time.Millisecond
-	storeCfg.RaftElectionTimeoutTicks = 2
 	currentTrigger := make(chan *roachpb.SplitTrigger, 1)
 	var seen struct {
 		syncutil.Mutex
@@ -1979,7 +2053,7 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 	}
 	seen.sids = make(map[kvserverbase.CmdIDKey][2]bool)
 
-	storeCfg.TestingKnobs.EvalKnobs.TestingEvalFilter = func(args kvserverbase.FilterArgs) *roachpb.Error {
+	testingEvalFilter := func(args kvserverbase.FilterArgs) *roachpb.Error {
 		et, ok := args.Req.(*roachpb.EndTxnRequest)
 		if !ok || et.InternalCommitTrigger == nil {
 			return nil
@@ -2013,16 +2087,37 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 		return nil
 	}
 
-	mtc.storeConfig = &storeCfg
-	defer mtc.Stop()
-	mtc.Start(t, 2)
+	ctx := context.Background()
+	tc := testcluster.StartTestCluster(t, 2,
+		base.TestClusterArgs{
+			ReplicationMode: base.ReplicationManual,
+			ServerArgs: base.TestServerArgs{
+				Knobs: base.TestingKnobs{
+					Store: &kvserver.StoreTestingKnobs{
+						EvalKnobs: kvserverbase.BatchEvalTestingKnobs{
+							TestingEvalFilter: testingEvalFilter,
+						},
+					},
+				},
+				RaftConfig: base.RaftConfig{
+					// An aggressive tick interval lets groups communicate more and thus
+					// triggers test failures much more reliably. We can't go too aggressive
+					// or race tests never make any progress.
+					RaftTickInterval:           100 * time.Millisecond,
+					RaftElectionTimeoutTicks:   2,
+					RaftHeartbeatIntervalTicks: 1,
+				},
+			},
+		})
+	defer tc.Stopper().Stop(ctx)
+	store := tc.GetFirstStoreFromServer(t, 0)
 
-	leftRange := mtc.stores[0].LookupReplica(roachpb.RKey("a"))
+	leftRange := store.LookupReplica(roachpb.RKey("a"))
 
 	// Replicate the left range onto the second node. We don't wait since we
 	// don't actually care what the second node does. All we want is that the
 	// first node isn't surprised by messages from that node.
-	mtc.replicateRange(leftRange.RangeID, 1)
+	tc.AddVotersOrFatal(t, leftRange.Desc().StartKey.AsRawKey(), tc.Target(1))
 
 	for i := 0; i < 10; i++ {
 		errChan := make(chan *roachpb.Error)
@@ -2038,7 +2133,7 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 			// range).
 			splitKey := roachpb.Key(encoding.EncodeVarintDescending([]byte("a"), int64(i)))
 			splitArgs := adminSplitArgs(splitKey)
-			_, pErr := kv.SendWrapped(context.Background(), mtc.distSenders[0], splitArgs)
+			_, pErr := kv.SendWrapped(context.Background(), tc.Servers[0].DistSender(), splitArgs)
 			errChan <- pErr
 		}()
 		go func() {
@@ -2059,7 +2154,7 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 			// side in the split trigger was racing with the uninitialized
 			// version for the same group, resulting in clobbered HardState).
 			for term := uint64(1); ; term++ {
-				if sent := mtc.transport.SendAsync(&kvserver.RaftMessageRequest{
+				if sent := tc.Servers[1].RaftTransport().SendAsync(&kvserver.RaftMessageRequest{
 					RangeID:     trigger.RightDesc.RangeID,
 					ToReplica:   replicas[0],
 					FromReplica: replicas[1],
@@ -2096,47 +2191,55 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 func TestLeaderAfterSplit(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeConfig := kvserver.TestStoreConfig(nil)
-	storeConfig.TestingKnobs.DisableReplicateQueue = true
-	storeConfig.TestingKnobs.DisableMergeQueue = true
-	storeConfig.RaftElectionTimeoutTicks = 1000000
-	mtc := &multiTestContext{
-		storeConfig: &storeConfig,
-	}
-	defer mtc.Stop()
-	mtc.Start(t, 3)
 
-	mtc.replicateRange(1, 1, 2)
+	ctx := context.Background()
+	tc := testcluster.StartTestCluster(t, 3,
+		base.TestClusterArgs{
+			ReplicationMode: base.ReplicationManual,
+			ServerArgs: base.TestServerArgs{
+				RaftConfig: base.RaftConfig{
+					RaftElectionTimeoutTicks: 1000000,
+				},
+			},
+		})
+	defer tc.Stopper().Stop(ctx)
+	store := tc.GetFirstStoreFromServer(t, 0)
 
 	leftKey := roachpb.Key("a")
 	splitKey := roachpb.Key("m")
 	rightKey := roachpb.Key("z")
 
+	repl := store.LookupReplica(roachpb.RKey(leftKey))
+	require.NotNil(t, repl)
+	tc.AddVotersOrFatal(t, repl.Desc().StartKey.AsRawKey(), tc.Targets(1, 2)...)
+
 	splitArgs := adminSplitArgs(splitKey)
-	if _, pErr := kv.SendWrapped(context.Background(), mtc.distSenders[0], splitArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, tc.Servers[0].DistSender(), splitArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
 	incArgs := incrementArgs(leftKey, 1)
-	if _, pErr := kv.SendWrapped(context.Background(), mtc.distSenders[0], incArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, tc.Servers[0].DistSender(), incArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
 	incArgs = incrementArgs(rightKey, 2)
-	if _, pErr := kv.SendWrapped(context.Background(), mtc.distSenders[0], incArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, tc.Servers[0].DistSender(), incArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 }
 
 func BenchmarkStoreRangeSplit(b *testing.B) {
-	var mtc multiTestContext
-	mtc.Start(b, 1)
-	defer mtc.Stop()
-	store := mtc.Store(0)
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(b, base.TestServerArgs{})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(b, err)
 
 	// Perform initial split of ranges.
 	sArgs := adminSplitArgs(roachpb.Key("b"))
-	if _, err := kv.SendWrapped(context.Background(), store.TestSender(), sArgs); err != nil {
+	if _, err := kv.SendWrapped(ctx, store.TestSender(), sArgs); err != nil {
 		b.Fatal(err)
 	}
 
@@ -2148,7 +2251,7 @@ func BenchmarkStoreRangeSplit(b *testing.B) {
 
 	// Merge the b range back into the a range.
 	mArgs := adminMergeArgs(roachpb.KeyMin)
-	if _, err := kv.SendWrapped(context.Background(), store.TestSender(), mArgs); err != nil {
+	if _, err := kv.SendWrapped(ctx, store.TestSender(), mArgs); err != nil {
 		b.Fatal(err)
 	}
 
@@ -2156,13 +2259,13 @@ func BenchmarkStoreRangeSplit(b *testing.B) {
 	for i := 0; i < b.N; i++ {
 		// Split the range.
 		b.StartTimer()
-		if _, err := kv.SendWrapped(context.Background(), store.TestSender(), sArgs); err != nil {
+		if _, err := kv.SendWrapped(ctx, store.TestSender(), sArgs); err != nil {
 			b.Fatal(err)
 		}
 
 		// Merge the ranges.
 		b.StopTimer()
-		if _, err := kv.SendWrapped(context.Background(), store.TestSender(), mArgs); err != nil {
+		if _, err := kv.SendWrapped(ctx, store.TestSender(), mArgs); err != nil {
 			b.Fatal(err)
 		}
 	}
@@ -2224,22 +2327,30 @@ func writeRandomTimeSeriesDataToRange(
 func TestStoreRangeGossipOnSplits(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.GossipWhenCapacityDeltaExceedsFraction = 0.5 // 50% for testing
-	// We can't properly test how frequently changes in the number of ranges
-	// trigger the store to gossip its capacities if we have to worry about
-	// changes in the number of leases also triggering store gossip.
-	storeCfg.TestingKnobs.DisableLeaseCapacityGossip = true
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	storeCfg.TestingKnobs.DisableScanner = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue:                      true,
+				DisableSplitQueue:                      true,
+				DisableScanner:                         true,
+				GossipWhenCapacityDeltaExceedsFraction: 0.5, // 50% for testing
+				// We can't properly test how frequently changes in the number of ranges
+				// trigger the store to gossip its capacities if we have to worry about
+				// changes in the number of leases also triggering store gossip.
+				DisableLeaseCapacityGossip: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 	storeKey := gossip.MakeStoreKey(store.StoreID())
 
 	// Avoid excessive logging on under-replicated ranges due to our many splits.
-	config.TestingSetupZoneConfigHook(stopper)
+	config.TestingSetupZoneConfigHook(s.Stopper())
 	zoneConfig := zonepb.DefaultZoneConfig()
 	zoneConfig.NumReplicas = proto.Int32(1)
 	config.TestingSetZoneConfig(0, zoneConfig)
@@ -2308,16 +2419,24 @@ func TestStoreRangeGossipOnSplits(t *testing.T) {
 func TestStoreTxnWaitQueueEnabledOnSplit(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	key := keys.UserTableDataMin
 	args := adminSplitArgs(key)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), args); pErr != nil {
 		t.Fatalf("%q: split unexpected error: %s", key, pErr)
 	}
 
@@ -2332,17 +2451,25 @@ func TestStoreTxnWaitQueueEnabledOnSplit(t *testing.T) {
 func TestDistributedTxnCleanup(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.Background())
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+
+	ctx := context.Background()
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue: true,
+				DisableSplitQueue: true,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	// Split at "a".
 	lhsKey := roachpb.Key("a")
 	args := adminSplitArgs(lhsKey)
-	if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, store.TestSender(), args); pErr != nil {
 		t.Fatalf("split at %q: %s", lhsKey, pErr)
 	}
 	lhs := store.LookupReplica(roachpb.RKey("a"))
@@ -2366,7 +2493,6 @@ func TestDistributedTxnCleanup(t *testing.T) {
 		testutils.RunTrueAndFalse(t, "commit", func(t *testing.T, commit bool) {
 			// Run a distributed transaction involving the lhsKey and rhsKey.
 			var txnKey roachpb.Key
-			ctx := context.Background()
 			txn := kv.NewTxn(ctx, store.DB(), 0 /* gatewayNodeID */)
 			txnFn := func(ctx context.Context, txn *kv.Txn) error {
 				b := txn.NewBatch()
@@ -2526,10 +2652,7 @@ func TestTxnWaitQueueDependencyCycleWithRangeSplit(t *testing.T) {
 		var pushCount int32
 		firstPush := make(chan struct{})
 
-		storeCfg := kvserver.TestStoreConfig(nil)
-		storeCfg.TestingKnobs.DisableSplitQueue = true
-		storeCfg.TestingKnobs.DisableMergeQueue = true
-		storeCfg.TestingKnobs.EvalKnobs.TestingEvalFilter =
+		testingEvalFilter :=
 			func(filterArgs kvserverbase.FilterArgs) *roachpb.Error {
 				if _, ok := filterArgs.Req.(*roachpb.PushTxnRequest); ok {
 					if atomic.AddInt32(&pushCount, 1) == 1 {
@@ -2538,16 +2661,29 @@ func TestTxnWaitQueueDependencyCycleWithRangeSplit(t *testing.T) {
 				}
 				return nil
 			}
-		stopper := stop.NewStopper()
-		defer stopper.Stop(context.Background())
-		store := createTestStoreWithConfig(t, stopper, storeCfg)
+		ctx := context.Background()
+		serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					DisableMergeQueue: true,
+					DisableSplitQueue: true,
+					EvalKnobs: kvserverbase.BatchEvalTestingKnobs{
+						TestingEvalFilter: testingEvalFilter,
+					},
+				},
+			},
+		})
+		s := serv.(*server.TestServer)
+		defer s.Stopper().Stop(ctx)
+		store, err := s.Stores().GetStore(s.GetFirstStoreID())
+		require.NoError(t, err)
 
 		lhsKey := roachpb.Key("a")
 		rhsKey := roachpb.Key("b")
 
 		// Split at "a".
 		args := adminSplitArgs(lhsKey)
-		if _, pErr := kv.SendWrapped(context.Background(), store.TestSender(), args); pErr != nil {
+		if _, pErr := kv.SendWrapped(ctx, store.TestSender(), args); pErr != nil {
 			t.Fatalf("split at %q: %s", lhsKey, pErr)
 		}
 		lhs := store.LookupReplica(roachpb.RKey("a"))
@@ -2562,7 +2698,7 @@ func TestTxnWaitQueueDependencyCycleWithRangeSplit(t *testing.T) {
 		// Start txn to write key a.
 		txnACh := make(chan error)
 		go func() {
-			txnACh <- store.DB().Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+			txnACh <- store.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 				if err := txn.Put(ctx, lhsKey, "value"); err != nil {
 					return err
 				}
@@ -2579,7 +2715,7 @@ func TestTxnWaitQueueDependencyCycleWithRangeSplit(t *testing.T) {
 		// Start txn to write key b.
 		txnBCh := make(chan error)
 		go func() {
-			txnBCh <- store.DB().Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+			txnBCh <- store.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 				if err := txn.Put(ctx, rhsKey, "value"); err != nil {
 					return err
 				}
@@ -2611,7 +2747,7 @@ func TestTxnWaitQueueDependencyCycleWithRangeSplit(t *testing.T) {
 
 		// Split at "b".
 		args = adminSplitArgs(rhsKey)
-		if _, pErr := kv.SendWrappedWith(context.Background(), store, roachpb.Header{
+		if _, pErr := kv.SendWrappedWith(ctx, store, roachpb.Header{
 			RangeID: lhs.RangeID,
 		}, args); pErr != nil {
 			t.Fatalf("split at %q: %s", rhsKey, pErr)
@@ -2962,12 +3098,7 @@ func TestRangeLookupAsyncResolveIntent(t *testing.T) {
 	blockPushTxn := make(chan struct{})
 	defer close(blockPushTxn)
 
-	// Disable async tasks in the intent resolver. All tasks will be synchronous.
-	cfg := kvserver.TestStoreConfig(nil)
-	cfg.TestingKnobs.IntentResolverKnobs.ForceSyncIntentResolution = true
-	cfg.TestingKnobs.DisableSplitQueue = true
-	cfg.TestingKnobs.DisableMergeQueue = true
-	cfg.TestingKnobs.TestingProposalFilter =
+	testingProposalFilter :=
 		func(args kvserverbase.ProposalFilterArgs) *roachpb.Error {
 			for _, union := range args.Req.Requests {
 				if union.GetInner().Method() == roachpb.PushTxn {
@@ -2978,9 +3109,23 @@ func TestRangeLookupAsyncResolveIntent(t *testing.T) {
 			return nil
 		}
 	ctx := context.Background()
-	stopper := stop.NewStopper()
-	defer stopper.Stop(ctx)
-	store := createTestStoreWithConfig(t, stopper, cfg)
+	srv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				// Disable async tasks in the intent resolver. All tasks will be synchronous.
+				IntentResolverKnobs: kvserverbase.IntentResolverTestingKnobs{
+					ForceSyncIntentResolution: true,
+				},
+				DisableMergeQueue:     true,
+				DisableSplitQueue:     true,
+				TestingProposalFilter: testingProposalFilter,
+			},
+		},
+	})
+	s := srv.(*server.TestServer)
+	defer s.Stopper().Stop(context.Background())
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
 
 	// Split range 1 at an arbitrary key so that we're not dealing with the
 	// first range for the rest of this test. The first range is handled
@@ -3172,7 +3317,7 @@ func TestSplitTriggerMeetsUnexpectedReplicaID(t *testing.T) {
 	})
 
 	close(blockPromoteCh)
-	if err := g.Wait(); !kvserver.IsRetriableReplicationChangeError(err) {
+	if err := g.Wait(); !testutils.IsError(err, `descriptor changed`) {
 		t.Fatalf(`expected "descriptor changed" error got: %+v`, err)
 	}
 
@@ -3184,8 +3329,8 @@ func TestSplitTriggerMeetsUnexpectedReplicaID(t *testing.T) {
 		// has not heard a raft message addressed to a later replica ID while the
 		// "was not found on" error is expected if the store has heard that it has
 		// a newer replica ID before receiving the snapshot.
-		if !kvserver.IsRetriableReplicationChangeError(err) {
-			t.Fatal(err)
+		if !testutils.IsError(err, `snapshot intersects existing range|r[0-9]+ was not found on s[0-9]+`) {
+			t.Fatalf(`expected snapshot intersects existing range|r[0-9]+ was not found on s[0-9]+" error got: %+v`, err)
 		}
 	}
 	for i := 0; i < 5; i++ {
@@ -3265,14 +3410,21 @@ func TestSplitBlocksReadsToRHS(t *testing.T) {
 		return nil
 	}
 
-	storeCfg := kvserver.TestStoreConfig(nil)
-	storeCfg.TestingKnobs.DisableSplitQueue = true
-	storeCfg.TestingKnobs.DisableMergeQueue = true
-	storeCfg.TestingKnobs.TestingProposalFilter = propFilter
 	ctx := context.Background()
-	stopper := stop.NewStopper()
-	defer stopper.Stop(ctx)
-	store := createTestStoreWithConfig(t, stopper, storeCfg)
+	serv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &kvserver.StoreTestingKnobs{
+				DisableMergeQueue:     true,
+				DisableSplitQueue:     true,
+				TestingProposalFilter: propFilter,
+			},
+		},
+	})
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+
 	repl := store.LookupReplica(roachpb.RKey(keySplit))
 	tsBefore := store.Clock().Now()
 

--- a/pkg/kv/kvserver/client_test.go
+++ b/pkg/kv/kvserver/client_test.go
@@ -1663,6 +1663,9 @@ func verifyRangeStats(
 	if err != nil {
 		return err
 	}
+	// When used with a real wall clock these will not be the same, since it
+	// takes time to load stats.
+	expMS.AgeTo(ms.LastUpdateNanos)
 	// Clear system counts as these are expected to vary.
 	ms.SysBytes, ms.SysCount, ms.AbortSpanBytes = 0, 0, 0
 	if ms != expMS {

--- a/pkg/kv/kvserver/store.go
+++ b/pkg/kv/kvserver/store.go
@@ -735,11 +735,6 @@ type StoreConfig struct {
 	// HistogramWindowInterval is (server.Config).HistogramWindowInterval
 	HistogramWindowInterval time.Duration
 
-	// GossipWhenCapacityDeltaExceedsFraction specifies the fraction from the last
-	// gossiped store capacity values which need be exceeded before the store will
-	// gossip immediately without waiting for the periodic gossip interval.
-	GossipWhenCapacityDeltaExceedsFraction float64
-
 	// ExternalStorage creates ExternalStorage objects which allows access to external files
 	ExternalStorage        cloud.ExternalStorageFactory
 	ExternalStorageFromURI cloud.ExternalStorageFromURIFactory
@@ -791,8 +786,8 @@ func (sc *StoreConfig) SetDefaults() {
 			envutil.EnvOrDefaultInt("COCKROACH_CONCURRENT_SNAPSHOT_APPLY_LIMIT", 1)
 	}
 
-	if sc.GossipWhenCapacityDeltaExceedsFraction == 0 {
-		sc.GossipWhenCapacityDeltaExceedsFraction = defaultGossipWhenCapacityDeltaExceedsFraction
+	if sc.TestingKnobs.GossipWhenCapacityDeltaExceedsFraction == 0 {
+		sc.TestingKnobs.GossipWhenCapacityDeltaExceedsFraction = defaultGossipWhenCapacityDeltaExceedsFraction
 	}
 }
 
@@ -1919,9 +1914,9 @@ func (s *Store) GossipStore(ctx context.Context, useCached bool) error {
 	// the usual periodic interval. Re-gossip more rapidly for RangeCount
 	// changes because allocators with stale information are much more
 	// likely to make bad decisions.
-	rangeCountdown := float64(storeDesc.Capacity.RangeCount) * s.cfg.GossipWhenCapacityDeltaExceedsFraction
+	rangeCountdown := float64(storeDesc.Capacity.RangeCount) * s.cfg.TestingKnobs.GossipWhenCapacityDeltaExceedsFraction
 	atomic.StoreInt32(&s.gossipRangeCountdown, int32(math.Ceil(math.Min(rangeCountdown, 3))))
-	leaseCountdown := float64(storeDesc.Capacity.LeaseCount) * s.cfg.GossipWhenCapacityDeltaExceedsFraction
+	leaseCountdown := float64(storeDesc.Capacity.LeaseCount) * s.cfg.TestingKnobs.GossipWhenCapacityDeltaExceedsFraction
 	atomic.StoreInt32(&s.gossipLeaseCountdown, int32(math.Ceil(math.Max(leaseCountdown, 1))))
 	syncutil.StoreFloat64(&s.gossipQueriesPerSecondVal, storeDesc.Capacity.QueriesPerSecond)
 	syncutil.StoreFloat64(&s.gossipWritesPerSecondVal, storeDesc.Capacity.WritesPerSecond)

--- a/pkg/kv/kvserver/testing_knobs.go
+++ b/pkg/kv/kvserver/testing_knobs.go
@@ -266,6 +266,10 @@ type StoreTestingKnobs struct {
 	// If set, use the given truncated state type when bootstrapping ranges.
 	// This is used for testing the truncated state migration.
 	TruncatedStateTypeOverride *stateloader.TruncatedStateType
+	// GossipWhenCapacityDeltaExceedsFraction specifies the fraction from the last
+	// gossiped store capacity values which need be exceeded before the store will
+	// gossip immediately without waiting for the periodic gossip interval.
+	GossipWhenCapacityDeltaExceedsFraction float64
 }
 
 // ModuleTestingKnobs is part of the base.ModuleTestingKnobs interface.

